circle.yml

```diff
@@ -0,0 +1,23 @@
+machine:
+  timezone: UTC
+  java:
+    version: oraclejdk8
+  environment:
+    _JAVA_OPTIONS: "-Xms512m -Xmx1024m"
```

Comment:
I think using `2g` should be fine here. Travis CI and Circle CI allow to use 4 GB of memory per build job.